toolchain: gcc: fix build error with glibc for aarch64 targets

rockdrilla opened the pull request:
GCC 14+ fails to build due to libatomic specific -march handling.
This build error triggers only with glibc and not with musl libc which is default.

Closes: https://github.com/openwrt/openwrt/issues/16073

zhouF96 commented:
> @robimarko, @hauke, @rsalvaterra, friendly ping, PTAL.
> 
> @zhouF96 cherry-pick this MR and try again. :)

Thanks, this solves my problem!

robimarko commented on toolchain/gcc/patches-14.x/830-aarch64-libatomic.patch:
```diff
@@ -0,0 +1,46 @@
+Inspired by The Yocto Project [1]
+
+[1] https://github.com/yoctoproject/poky/blob/master/meta/recipes-devtools/gcc/gcc/0022-libatomic-Do-not-enforce-march-on-aarch64.patch
+
```
Please add your SoB to the patch as well

rockdrilla committed:
toolchain: gcc: fix build error with glibc for aarch64 targets

GCC 14+ fails to build due to libatomic specific -march handling.
This build error triggers only with glibc and not with musl libc
which is default.

Related: https://github.com/openwrt/openwrt/issues/16073

Signed-off-by: Konstantin Demin <rockdrilla@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/18647
Signed-off-by: Robert Marko <robimarko@gmail.com>

robimarko commented:
Thanks! Rebased on top of main and merged!
